x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryFolder.java

```diff
@@ -455,6 +455,13 @@ static EsQueryExec fold(AggregateExec a, EsQueryExec exec) {
                 // literal
                 if (target.foldable()) {
                     queryC = queryC.addColumn(ne.toAttribute());
+                    //If we're only selecting literals, we have to fix the row count
+                    if(a.aggregates().stream().allMatch(s -> s.children().get(0) instanceof Literal)) {
+                        for (Expression grouping : a.groupings()) {
```

Comment:
I don't think that's the correct place to add the groupings.
We are iterating over all `a.aggregates()`.

Reply:
yep, 100% true.